### PR TITLE
fixes broken model blueprint. Adds blueprint tests.

### PR DESCRIPTION
**Pull request by aars**
I think this fully fixes the broken model blueprint. I don't think it was related to dasherized model names, if anything dasherized project-names.

I'm pretty sure the `addon-import` is not to be generated with a `jsonapi-resource` and `originBlueprintName` property caused the `addon-import` blueprint to run over the generated model, breaking it.

I think. This was my first time diving into blueprints/ember-cli-addon's.

### CODE REVIEW

**Comment by pixelhandler** [comment]
@aars thats great to add blueprint tests!

File: blueprints/jsonapi-resource/index.js [hunk 1]
```diff
@@ -57,19 +57,16 @@ module.exports = {
...
-      },
-      originBlueprintName: 'jsonapi-model'
+      }
     });
 
     var otherOptions = merge({}, options);
 
     return Promise.all([
       this._processBlueprint(type, 'jsonapi-model', modelOptions),
-      this._processBlueprint(type, 'addon-import', modelOptions),
```
Comment (pixelhandler) [inline blueprints/jsonapi-resource/index.js]: The addon-import blueprint is the one the creates the import of the add-on module in the app directory. I think it's necessary.

File: blueprints/jsonapi-resource/index.js [hunk 2]
```diff
@@ -57,19 +57,16 @@ module.exports = {
...
       this._processBlueprint(type, 'jsonapi-adapter', otherOptions),
       this._processBlueprint(type, 'jsonapi-serializer', otherOptions),
       this._processBlueprint(type, 'jsonapi-service', otherOptions),
-      this._processBlueprint(type, 'jsonapi-adapter', otherOptions),
```
Comment (pixelhandler) [inline blueprints/jsonapi-resource/index.js]: Opps I had this in there twice, good catch.

**Comment by pixelhandler** [comment]
@aars see the addon-import file in ember-cli, https://github.com/ember-cli/ember-cli/blob/master/blueprints/addon-import/index.js it a wrapper the tests have an example, https://github.com/ember-cli/ember-cli/blob/master/tests/acceptance/addon-generate-test.js

File: blueprints/jsonapi-resource/index.js (hunk 1 above)
Comment (aars) [inline blueprints/jsonapi-resource/index.js]: I'm sure it's necessary, but not when generating a `jsonapi-resource`, right?

I'll give this some more attention/testing with your jr-test repo.

**Comment by aars** [comment]
@pixelhandler I'm not yet familiar with the whole ember-cli addon setup/structure. Will be soon. I'll do some more testing.

**Comment by pixelhandler** [comment]
@aars I can try this out on the jr-test app, https://github.com/pixelhandler/jr-test

**Comment by pixelhandler** [comment]
@aars I'll merge that and add another comment after as well.

**Pull request merged**